test/mock/createInsightsClient.ts

```diff
@@ -26,13 +26,23 @@ export function createAlgoliaAnalytics() {
       callback(values._userToken);
     }
   };
+  const sendEvent = () => {
```

Comment:
Why do we implement another whole logic in its mock?

Reply:
Good point. I thought about it again, and I think the new test is better.
https://github.com/algolia/instantsearch.js/pull/4575/commits/0add3dcab5e1f18e57e86fdcc6f00ed6bdc7e909

Reply:
We probably shouldn't even mock any logic, but just spy for methods called. That's coming from before so it shouldn't block that PR though.

Reply:
At some point, I began to wonder if I just should've used the real search-insights client for the tests, because there are some implementations on the search-insights side that needs to be tested with the insights middleware.